internal/statemachine/classic_states.go

```diff
@@ -1161,11 +1161,10 @@ func (stateMachine *StateMachine) preseedClassicImage() error {
 	var mountCmds []*exec.Cmd
 	var umountCmds []*exec.Cmd
 	for _, mountPoint := range mountPoints {
-		var mountCmd, umountCmd *exec.Cmd
-		mountCmd, umountCmd = mountFromHost(stateMachine.tempDirs.chroot, mountPoint)
-		defer umountCmd.Run()
-		mountCmds = append(mountCmds, mountCmd)
-		umountCmds = append(umountCmds, umountCmd)
+		mountCmds, umountCmds := mountFromHost(stateMachine.tempDirs.chroot, mountPoint)
+		defer runAll(umountCmds)
+		mountCmds = append(mountCmds, mountCmds...)
+		umountCmds = append(umountCmds, umountCmds...)
```

Comment:
This here is busted, sadly. As written right now, it causes both `umountCmds` and `mountCmds` to end up empty. You need to get the `mountFromHost()` cmd lists to some temporary variables and then `append()` them to the final ones. Basically same as it was before but with lists should be good!

Reply:
Doh!